packages/data/src/components/use-select/index.js

```diff
@@ -61,7 +62,16 @@ function Store( registry, suspense ) {
 			// in the interval between the `getValue` call during render and creating
 			// the subscription, which is slightly delayed. We need to ensure that this
 			// second `getValue` call will compute a fresh value.
-			lastMapResultValid = false;
+			lastMapResultValid =
+				activeStores.length === storeStatesOnMount.length &&
+				activeStores.every(
+					( storeName, index ) =>
+						storeStatesOnMount[ index ] ===
+						// To do: figure out why it's sometimes not available in
+						// unit tests.
+						registry.stores[ storeName ]?.store?.getState?.()
```

Comment:
@jsnajdr @youknowriad Any idea why this is not available in some unit tests? Something about custom generic stores? Is there a better way to may store names to their state?

Reply:
It should be, the graph looks the same as the other optimisation

Reply:
Getting -6.75% for first block load in this run.

Reply:
I guess if there's no interface, we can always return false for now (previous behaviour). But strange that there's no way to get the state.

Reply:
Yes, if you encounter a generic store, just bail out of the optimization. They are very rare anyway.

> But strange that there's no way to get the state.

There doesn't need to be any state, it doesn't have to be a Redux store at all. Just an object that satisfies a generic interface: expose selectors to get the current data, and a `subscribe` function to subscribe to updates. Something like `Observable`.

Reply:
Ok, done. Returning an empty object will work because the equality check will fail.

Reply:
I'm thinking we could add an optional `getSnapshot` or `getStableReference` to the interface to address this use-case.

Reply:
Should we try that separately?

Reply:
Fine with it.

Reply:
Second run after the rewrite is 6.6% for first block load

Reply:
> I'm thinking we could add an optional `getSnapshot` or `getStableReference` to the interface

Yes, and to prevent misuse to read private-ish data, return an empty object `{}` memoized/keyed in weakmap by the real state object.